src/metabase/search/ingestion.clj

```diff
@@ -115,24 +115,24 @@
 
 (defn consume!
   "Update all active engines' indexes with the given documents"
-  [documents-reducible]
+  [documents-reducible re-indexing?]
   (when-let [engines (seq (search.engine/active-engines))]
     (if (= 1 (count engines))
-      (search.engine/consume! (first engines) documents-reducible)
+      (search.engine/consume! (first engines) documents-reducible re-indexing?)
       ;; TODO um, multiplexing over the reducible awkwardly feels strange. We at least use a magic number for now.
       (doseq [batch (eduction (partition-all 150) documents-reducible)
               e     engines]
-        (search.engine/consume! e batch)))))
+        (search.engine/consume! e batch re-indexing?)))))
 
 (defn bulk-ingest!
   "Process the given search model updates. Returns the number of search index entries that get updated as a result."
-  [updates]
+  [updates re-indexing?]
   (->> (for [[search-model where-clauses] (u/group-by first second updates)]
          (spec-index-reducible search-model (into [:or] (distinct where-clauses))))
        ;; init collection is only for clj-kondo, as we know that the list is non-empty
        (reduce u/rconcat [])
        query->documents
-       consume!))
+       (#(consume! % re-indexing?))))
```

Comment:
Oh man I hate having to do this kind of double nesting.

Thinking whether it could make sense to swap the args around because the implementation of consume also find the order awkward. Feels like a smell to not have the list argument last if its used in "pipelines" like this.

Rather than `re-indexing?` flag you could make have a `context` keyword argument, with values like `:indexing`, `:re-indexing`, and `:update`? Taking that first feels natural to me.